Python/ceval.c

```diff
@@ -863,6 +863,75 @@ static const binaryfunc binary_ops[] = {
     [NB_INPLACE_XOR] = PyNumber_InPlaceXor,
 };
 
+#define NEXT_OP_STORES(O)                                  \
```

Comment:
I don't like this. It adds even more branches to an already lengthy list of deopt checks, especially for the `int` forms.
https://github.com/python/cpython/blob/main/Python/adaptive.md#choice-of-specializations 

Most of this can be done at specialization time.